modules/ejb/src/main/java/org/jboss/weld/module/ejb/EjbSupportImpl.java

```diff
@@ -70,13 +73,25 @@ public void cleanup() {
         InternalEjbDescriptor<T> descriptor = InternalEjbDescriptor.of(d);
         EnhancedAnnotatedType<T> implementationClass = SessionBeans.getEjbImplementationClass(descriptor, manager, type);
 
-        AbstractInstantiator<T> instantiator = null;
+        Instantiator<T> instantiator = null;
         if (type.equals(implementationClass)) {
             instantiator = new DefaultInstantiator<T>(type, null, manager);
         } else {
             // Session bean subclassed by the EJB container
             instantiator = SubclassedComponentInstantiator.forSubclassedEjb(type, implementationClass, null, manager);
         }
+        InterceptionModel interceptionModel = manager.getInterceptorModelRegistry().get(type.slim());
+        if (interceptionModel != null) {
+            if (interceptionModel.hasExternalNonConstructorInterceptors()) {
+                instantiator = SubclassedComponentInstantiator
+                        .forInterceptedDecoratedBean(implementationClass, null, (AbstractInstantiator<T>) instantiator, manager);
+                instantiator = new InterceptorApplyingInstantiator<>(instantiator, interceptionModel, type.slim());
+
+            }
+            if (interceptionModel.hasExternalConstructorInterceptors()) {
```

Comment:
I wonder if `@AroundConstruct` makes any sense for MDB? We don't control it's lifecycle and don't really know what EJB container chooses to do with it and when. But perhaps I am missing something?